consensus/hotstuff/cruisectl/block_rate_controller_test.go

```diff
@@ -0,0 +1,25 @@
+package cruisectl
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/module/irrecoverable"
+	mockprotocol "github.com/onflow/flow-go/state/protocol/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func TestStartStop(t *testing.T) {
+	state := mockprotocol.NewState(t)
+	ctl, err := NewBlockRateController(unittest.Logger(), DefaultConfig(), state)
+	require.NoError(t, err)
+
+	ctx, cancel := irrecoverable.NewMockSignalerContextWithCancel(t, context.Background())
+	ctl.Start(ctx)
+	unittest.AssertClosesBefore(t, ctl.Ready(), time.Second)
```

Comment:
I would generally prefer using require in such cases since it doesn't make sense to continue test if invariant is broken.

Reply:
https://github.com/onflow/flow-go/pull/4332/commits/d92dfad2c22ec50ee493dfd5aaa84be0eb371233